contracts/Registry.sol

```diff
@@ -31,10 +31,7 @@ contract Registry {
 
     function addVault(address vault) external onlyTokenHolders {
         address assetAddress = IVault(vault).asset();
-        require(
-            IVault(_vaultByAsset[assetAddress]).asset() == address(0),
-            "addVault: Vault with this asset token already added"
-        );
+        require(_vaultByAsset[assetAddress] == address(0), "addVault: Vault with this asset token already added");
```

Comment:
Maybe require `vault != address(0)` so that you can't push to the array by accident?

Reply:
I think that would just throw when you tried to get the asset, so I don think it's necessary

Reply:
I've added it anyway to be explicit